### DISCUSSION TITLE
structure mechanic analysis using moose

### DISCUSSION THREAD

**Discussion by tjumsl** [idaholab/moose#20285] [Q&A Modules: General] [answered]
I try to find an alternative structure analysis application to the commercial software NASTRAN. Moose seems to be very promising.

My main analysis area is using plate/shell/beam elements for airframe structures, including linear / non-linear / buckling / modal anlaysis / frequency response etc. I asssume Moose should be able to do this, is that correct?

**Comment by GiudGiud** [comment]
Hello

We dont have modal analysis
We have some capabilities for frequency response though they may be in MASTODON, @cbolisetti will know the most about this. 

Guillaume

**Comment by ats5662** [comment]
If MOOSE doesn't have what you're looking for try [openCFS](https://opencfs.org/index.html)

**Reply by tjumsl** [reply]
> If MOOSE doesn't have what you're looking for try [openCFS](https://opencfs.org/index.html)

Thanks for the kind suggestion. For OpenCFS, do you know whehter it can conver the structure areas I mentioned above?

I quickly checked the website, it seems the tutorial examples are mainly on acoustic aspects.

**Reply by ats5662** [reply]
Yeah, I use openCFS for acoustic simulations, so I am not very familiar with its structural dynamics capabilities. I know they exist though - check out these resources

[openCFS YouTube tutorials](https://www.youtube.com/channel/UC0Tj64YopXzC3MmwY3huSlA/videos)
[Cantilever beam eigenfrequency example](https://opencfs.gitlab.io/userdocu/Applications/Singlefield/Mechanics/Cantilever/#eigenfrequency-analysis)
[Singlefield mechanics testsuit](https://gitlab.com/openCFS/Testsuite/-/tree/master/TESTSUIT/Singlefield/Mechanics)

**Reply by tjumsl** [reply]
> Yeah, I use openCFS for acoustic simulations, so I am not very familiar with its structural dynamics capabilities. I know they exist though - check out these resources
> 
> [openCFS YouTube tutorials](https://www.youtube.com/channel/UC0Tj64YopXzC3MmwY3huSlA/videos) [Cantilever beam eigenfrequency example](https://opencfs.gitlab.io/userdocu/Applications/Singlefield/Mechanics/Cantilever/#eigenfrequency-analysis) [Singlefield mechanics testsuit](https://gitlab.com/openCFS/Testsuite/-/tree/master/TESTSUIT/Singlefield/Mechanics)

Thanks, I will check the materials and reach the openCFS team for further confirmation.

**Reply by oldninja** [reply]
Another one is [CalculiX ](http://www.dhondt.de/). The only issue is that plate/shell/beam elements are expanded internally into brick elements. But it can do linear, non-linear (material and contact), buckling, modal, and freq. response.

**Comment by cbolisetti** [comment, marked as answer]
@tjumsl, you can use MOOSE for several mechanics applications but it's still not at the extent of traditional structural engineering software. We have plate/beam/shell elements, but we cannot model nonlinearities yet. Also, we don't have Eigen value analysis merged into MOOSE. If you are looking to use solid elements or stay in the linear range, MOOSE is great for large problems and multiphysics applications.

**Reply by tjumsl** [reply]
> @tjumsl, you can use MOOSE for several mechanics applications but it's still not at the extent of traditional structural engineering software. We have plate/beam/shell elements, but we cannot model nonlinearities yet. Also, we don't have Eigen value analysis merged into MOOSE. If you are looking to use solid elements or stay in the linear range, MOOSE is great for large problems and multiphysics applications.

OK, i see. Many thanks for the kind response. How about XFEM crack propergation? Is it possible for MOOSE to carry out complicate crack propergation analysises? Solid elements will be used in most cases.

**Reply by jiangwen84** [reply]
What do you mean by complicated crack propagation?  MOOSE XFEM can do certain 2D and 3D crack propagation modeling. Branching or merging is not well supported.

**Reply by tjumsl** [reply]
> What do you mean by complicated crack propagation? MOOSE XFEM can do certain 2D and 3D crack propagation modeling. Branching or merging is not well supported.

I am trying to use XFEM as an alternative tools to the commercial software Franc3D, do you think this would be possible?

**Reply by jiangwen84** [reply]
Does Franc3D support XFEM? If you are comfortable with writing some codes, MOOSE will provide you much more flexibility. If you are a primary commercial code user, I guess Franc3D would be a better choice.

**Reply by tjumsl** [reply]
> Does Franc3D support XFEM? If you are comfortable with writing some codes, MOOSE will provide you much more flexibility. If you are a primary commercial code user, I guess Franc3D would be a better choice.

Franc3D ultilized traditional FEM methods which needs a lot of meshing work. I am willing to write some codes and will dig into more details of MOOSE XFEM.
